bin/sam-coverage-depth.py

```diff
@@ -180,81 +205,144 @@ def getReferenceSeq(fasta: str | None, referenceId: str) -> str | None:
 
 
 def processSubsection(
-    args: argparse.Namespace,
+    filename: str,
     start: int,
+    window: int,
+    minDepth: int | None,
+    maxDepth: int,
+    minSiteMutationRate: float | None,
+    collectOffsets: bool,
+    verbosity: int,
     referenceId: str,
     referenceSeq: str | None,
-    returnCSV: bool,
+    returnList: bool,
     startTime: float,
-):
+) -> dict[str, Any]:
```

Comment:
I learned today that you can use `typing.TypedDict` to properly type dicts of which you know the keys.